ffenv/env.go

```diff
@@ -0,0 +1,57 @@
+package ffenv
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+// Parser is a parser for .env file format: flag=value. Each
+// line is tokenized as a single key/value pair.
+func Parser(r io.Reader, set func(name, value string) error) error {
+	return parse("", r, set)
+}
+
+func parse(prefix string, r io.Reader, set func(name, value string) error) error {
+	s := bufio.NewScanner(r)
+	for s.Scan() {
+		line := strings.TrimSpace(s.Text())
+		if line == "" {
+			continue // skip empties
+		}
+
+		if line[0] == '#' {
+			continue // skip comments
+		}
+
+		line = strings.TrimPrefix(line, prefix+"_")
+		var (
+			name  string
+			value string
+			index = strings.IndexRune(line, '=')
+		)
+		if index < 0 {
+			name, value = line, "true" // boolean option
+		} else {
+			name, value = strings.ToLower(line[:index]), line[index+1:]
+			name = strings.ReplaceAll(name, "_", "-")
```

Comment:
It's not a given that a `_` in an env var should match to only a `-` in a flag name. Quoting from [the docs](https://pkg.go.dev/github.com/peterbourgon/ff/v3@v3.0.0?tab=doc#WithEnvVarNoPrefix)

> Flag names are matched to environment variables by capitalizing the flag name, and replacing separator characters like periods or hyphens with underscores.

We would need the same kind of matching behavior here.

Reply:
So for each separator in `[]string{"-", ".", "/"}`, I should check that a flag exist with that format, or should I set it for each separator?
So given: `MY_FOO_LISTEN_ADDR=value` with prefix `MY_FOO`, I should set `listen.addr`, `listen-addr` and `listen/addr`, hoping there is one corresponding, and there is no conflict either.
Or iterate through existing flags (how in this context?) and do the other way around (like in the ff.Parse func)?

Reply:
I'm not sure what the correct implementation is, my point is that the behavior here should try to be consistent with env vars specified directly. If you're stuck, let me know, and I can dig in to the details and make a more direct suggestion.

Reply:
I went on the path where the option needs a `flag.FlagSet` passed, so it can match with the env var.
Concerning #69 , when the env var from a .env file is applied, it will apply itself on every flag. 
So, `FOO_VAR` will get applied to `-foo_var`, `-foo-var`, `-foo.var` and `-foo/var`. When using purely the flags, it can determine exactly which to update.
The updated code has been pushed in the branch now.